### PR TITLE
Make `emitFile` ensure the folders exists prior to writing to path

### LINKED ISSUE #1178: Emitter Utils emitFile should automatically create the parent folder if not there
It is a bit annoying to have to check the parent folder to the file you want to emit exists

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #1178.

This is just a port of what I did in Cadl APIView.

### CODE REVIEW

File: packages/compiler/core/emitter-utils.ts [hunk 1]
```diff
@@ -13,6 +15,13 @@ export interface EmitFileOptions {
  * @param options File Emitter options
  */
 export async function emitFile(program: Program, options: EmitFileOptions): Promise<void> {
+  // ensure path exists
+  const outputFolder = path.dirname(options.path);
+  try {
+    await mkdirp(outputFolder);
```
Comment (nguerrera) [inline packages/compiler/core/emitter-utils.ts:21]: This should go through the host. The catch is masking the issue. It should work with the virtual test file system. Perhaps we need to add mkdirp to the host interface.

Comment (nguerrera) [reply packages/compiler/core/emitter-utils.ts:21]: It's possible that it wouldn't fail and a test would start making real directories like this even though it's using a virtual file system. There shouldn't be any file IO done at all without going through the host interface.

File: packages/compiler/core/emitter-utils.ts [hunk 2]
```diff
@@ -13,6 +15,9 @@ export interface EmitFileOptions {
  * @param options File Emitter options
  */
 export async function emitFile(program: Program, options: EmitFileOptions): Promise<void> {
+  // ensure path exists
+  const outputFolder = path.dirname(options.path);
```
Comment (nguerrera) [inline packages/compiler/core/emitter-utils.ts:19]: I think this should use getDirectoryPath from our own path-utils. Sorry I missed that on first review.

File: packages/compiler/core/emitter-utils.ts [hunk 3]
```diff
@@ -1,4 +1,6 @@
 import { Program } from "./program.js";
+import path from "path";
+import mkdirp from "mkdirp";
```
Comment (nguerrera) [inline packages/compiler/core/emitter-utils.ts:3]: Remove unused imports.

**Comment by tjprescott** [comment]
@nguerrera thanks for the review. All comments should be addressed now.

**Review by nguerrera — APPROVED** [review]

**Pull request merged**